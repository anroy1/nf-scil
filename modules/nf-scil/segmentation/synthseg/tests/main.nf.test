nextflow_process {

    name "Test Process SEGMENTATION_SYNTHSEG"
    script "../main.nf"
    process "SEGMENTATION_SYNTHSEG"

    tag "modules"
    tag "modules_nfcore"
    tag "segmentation"
    tag "segmentation/synthseg"

    tag "subworkflows"
    tag "subworkflows/load_test_data"

    setup {
        run("LOAD_TEST_DATA", alias: "LOAD_DATA") {
            script "../../../../../subworkflows/nf-scil/load_test_data/main.nf"
            process {
                """
                input[0] = Channel.from( [ "freesurfer.zip" ] )
                input[1] = "test.load-test-data"
                """
            }
        }
    }

    test("segmentation - synthseg - basic") {
        config "./nextflow_basic.config"
        when {
            process {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ], // meta map
                        file("\${test_data_directory}/t1.nii.gz", checkIfExists: true,),
                        [],
                        file("\${test_data_directory}/license.txt", checkIfExists: true,)
                    ]
                }
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    Nifti_md5sum(process.out.wm_mask.get(0).get(1)),
                    Nifti_md5sum(process.out.gm_mask.get(0).get(1)),
                    Nifti_md5sum(process.out.csf_mask.get(0).get(1)),
                    process.out.versions
                ).match() }
            )
        }
    }

    test("segmentation - synthseg - options") {
        config "./nextflow_options.config"
        when {
            process {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ], // meta map
                        file("\${test_data_directory}/t1.nii.gz", checkIfExists: true,),
                        [],
                        file("\${test_data_directory}/license.txt", checkIfExists: true,)
                    ]
                }
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    Nifti_md5sum(process.out.wm_mask.get(0).get(1)),
                    Nifti_md5sum(process.out.gm_mask.get(0).get(1)),
                    Nifti_md5sum(process.out.csf_mask.get(0).get(1)),
                    file(process.out.vol.get(0).get(1)),
                    file(process.out.qc.get(0).get(1)),
                    // Nifti_md5sum(process.out.resample.get(0).get(1)),
                    process.out.versions
                ).match() }
            )
        }
    }

    // test("segmentation - synthseg - lesion") {
    //     config "./nextflow_lesion.config"
    //     when {
    //         process {
    //             """
    //             input[0] = LOAD_DATA.out.test_data_directory.map{
    //                 test_data_directory -> [
    //                     [ id:'test', single_end:false ], // meta map
    //                     file("\${test_data_directory}/t1.nii.gz", checkIfExists: true,),
    //                     file("\${test_data_directory}/lesion.nii.gz", checkIfExists: true,),
    //                     file("\${test_data_directory}/license.txt", checkIfExists: true,)

    //                 ]
    //             }
    //             """
    //         }
    //     }

    //     then {
    //         assertAll(
    //             { assert process.success },
    //             { assert snapshot(process.out).match() }
    //         )
    //     }
    // }

}
