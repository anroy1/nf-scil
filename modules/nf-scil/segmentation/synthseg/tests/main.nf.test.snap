{
    "segmentation - synthseg - options": {
        "content": [
            "test__mask_wm.nii.gz:md5,42ac2ae7eab32edb16f230211e5e55c5",
            "test__mask_gm.nii.gz:md5,7a205f13a6894200f0e1a4ea94447390",
            "test__mask_csf.nii.gz:md5,7b6125a133adbaae10e94d95782c6678",
            "test__vol.csv:md5,fe0078439655e81ab5777c73b4bd1edf",
            "test__qc.csv:md5,ede51ac9221945293200793a46c15afa",
            [
                "versions.yml:md5,b65418e9ce2ab1cce590b1f659e18c1d"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0-rc1",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-16T19:58:49.946322"
    },
    "segmentation - synthseg - basic": {
        "content": [
            "test__mask_wm.nii.gz:md5,f35c2cd0f899d5a290083d09bac01670",
            "test__mask_gm.nii.gz:md5,ef1aef14d90f78dcae60f3f73066af53",
            "test__mask_csf.nii.gz:md5,fa08d8c09176d72b8f9b3667419c3d3e",
            [
                "versions.yml:md5,b65418e9ce2ab1cce590b1f659e18c1d"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0-rc1",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-16T19:58:09.501194"
    }
}